gene_gene/gene_interacts_gene/dataset_statistics/dataset_statistics.py

```diff
@@ -0,0 +1,520 @@
+
+# In[ ]:
+
+
+sql= '''
+select id as sentence_id, text, (
+    char_length(regexp_replace(CAST(words AS TEXT), '[\u0080-\u00ff]', '', 'g')) - 
+    char_length(regexp_replace(regexp_replace(CAST(words AS TEXT), '[\u0080-\u00ff]', '', 'g'), ',', '','g'))
+) as sen_length, entity_types 
+from sentence 
+where entity_types::text like '%%Gene%%' or entity_types::text like '%%Gene%%';
+'''
+sentence_df = pd.read_sql(sql, database_str)
+sentence_df.head(2)
+
+
+# In[ ]:
+
+
+entity_data = []
+tagging_error_ids = set({})
+
+#skip tagging error
+skip_tag_error = False
+for index, row in tqdm_notebook(sentence_df.iterrows()):
+    
+    #create dictionay for mapping entity types
+    entity_mapper = {"sentence_id": row['sentence_id']}
+    
+    #Keep track of previous entity
+    previous_entity = 'o'
+    
+    #For all entitys in a given sentence decide what is tagged
+    for entity in row['entity_types']:
+        entity = entity.lower()
+
+        #Non-O tag
+        if entity != 'o' and previous_entity =='o':
+            #If entity not seen before instanciate it
+            if entity not in entity_mapper:
+                entity_mapper[entity] =0
+            entity_mapper[entity] += 1
+
+        # If previous tag was non-O and the current tag does not equal previous
+        # Then tagging error. e.x. Disease, Gene, Disease instead of Disease, O, Disease
+        elif entity != previous_entity and entity != 'o':
+            tagging_error_ids.add(row['sentence_id'])
+            skip_tag_error = True
+            break
+
+        previous_entity = entity
+    
+    # Do not add errors to dataframe
+    # They will be thrown out
+    if not skip_tag_error:
+        entity_data.append(entity_mapper)
+
+    skip_tag_error=False
+    
+entity_stats_df = pd.DataFrame.from_dict(entity_data).fillna(0)
+entity_stats_df.head(2)
+
+
+# In[ ]:
+
+
+tagging_error_df = pd.Series(sorted(list(tagging_error_ids)))
+tagging_error_df.to_csv("data/tagging_error_ids.tsv.xz", sep="\t",  index=False, compression="xz")
+tagging_error_df.head(2)
+
+
+# In[ ]:
+
+
+print(
+    "Total Number of IOB Tagging Errors: {}. Percentage of sentences affected: {:.2f}".format(
+        tagging_error_df.shape[0],
+        100*tagging_error_df.shape[0]/sentence_df.shape[0]
+    )
+)
+
+
+# In[ ]:
+
+
+header = ["sentence_id", "text", "sen_length"]
+sentence_df[header].to_csv("data/sentence_stats.tsv.xz", sep="\t", index=False, compression="xz")
+entity_stats_df.to_csv("data/entity_stats.tsv.xz", sep="\t",  index=False, compression="xz")
+
+
+# # Sentence Counts and Statistics
+
+# Below is the block of code that contains information about the full distribution of sentences tied to each candidate pair. Multiple sentences can contain more than one co-occuring pair, which results in some sentences being counted more than once.
```

Comment:
Maybe add an example of what you mean when you say having multiple pairs.  I at least remember I was confused by this before we talked.